checks.d/http_check.py

```diff
@@ -262,6 +262,17 @@ def _check(self, instance):
             running_time = time.time() - start
             # Store tags in a temporary list so that we don't modify the global tags data structure
             tags_list = list(tags)
+
+            # Check for ill-formatted tags that aren't string types and log an error
+            bad_tags = []
+            for tag in tags_list:
+                if type(tag) is not str:
+                    bad_tags.append(tag)
```

Comment:
So up for debate here is what should we do with the bad tags? Should we remove bad tags from the tag list - log the broken tags and submit the metric with the OK tags. Or should we identify the broken tags, log them and not send anything? We're currently identifying the "broken" tags, and submitting `tags_list` regardless of the fact it might contain bad tags. So we should either not make the call, or make the call without the broken tags.

I personally think we should only attempt the submission if there are no broken tags, to guarantee the customer _knows_ that if his check is working it's because the config is fine (and not partly fine).